upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
```

Comment:
**Code Quality and Style**: Consider extracting the `XMLHttpRequest` setup into a separate function to reduce redundancy and make the code more modular. This will also help in managing connections more efficiently.

**Correctness**: Ensure that each chunk's upload state is independently managed in case of failures or retries.

**Error Handling**: Implement a retry mechanism for network errors and utilize proper error handling instead of just `console.log`. Consider using Promises or the Fetch API for better async handling than `XMLHttpRequest`.

**Performance**: Creating a new `XMLHttpRequest` for every chunk is inefficient. Consider using a single connection or batching multiple chunks together when possible to optimize network usage.

To improve, I'd suggest using the Fetch API along with async/await for a cleaner and more efficient upload process.